pkg/proto/hint/hint.go

```diff
@@ -0,0 +1,173 @@
+package hint
+
+import (
+	"bufio"
+	"bytes"
+	"strings"
+)
+
+import (
+	"github.com/pkg/errors"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/runtime/misc"
+)
+
+const (
+	_            Type = iota
+	TypeMaster        // force route to master node
+	TypeSlave         // force route to slave node
+	TypeRoute         // custom route
+	TypeFullScan      // enable full-scan
+	TypeDirect        // direct route
+)
+
+var _hintTypes = [...]string{
+	TypeMaster:   "MASTER",
+	TypeSlave:    "SLAVE",
+	TypeRoute:    "ROUTE",
+	TypeFullScan: "FULLSCAN",
+	TypeDirect:   "DIRECT",
+}
+
+// KeyValue represents a pair of key and value.
+type KeyValue struct {
+	K string // key (optional)
+	V string // value
+}
+
+// Type represents the type of Hint.
+type Type uint8
+
+// String returns the display string.
+func (tp Type) String() string {
+	return _hintTypes[tp]
+}
+
+// Hint represents a Hint, a valid Hint should include type and input kv pairs.
+//
+// Follow the format below:
+//   - without inputs: YOUR_HINT()
+//   - with non-keyed inputs: YOUR_HINT(foo,bar,quz)
+//   - with keyed inputs: YOUR_HINT(x=foo,y=bar,z=quz)
+//
+type Hint struct {
+	Typ    Type
```

Comment:
why the variable name is `Typ`?

Reply:
Just a personal hobby, will be renamed to `Type`.